scrunch/datasets.py

```diff
@@ -1394,6 +1374,14 @@ def add_filter(self, name, expr, public=False):
         new_filter = self.resource.filters.create(payload)
         return self.filters[new_filter.body['name']]
 
+    def add_deck(self, name, description="", public=False):
+        payload = dict(element='shoji:entity',
+                       body=dict(name=name,
+                                 description=description,
+                                 is_public=public))
+        new_deck = self.resource.decks.create(payload)
+        return self.decks[new_deck.body['name']]
```

Comment:
You're fetching the deck by name here, but under `.decks` they are keyed by `.id`. This will KeyError.
There is no constrain for deck names to be unique so it cannot be used as a key value.